### ISSUE TITLE
Can't load pg gem with TruffleRuby 19.3 master builds

### ISSUE THREAD

**Issue by nirvdrum** [oracle/truffleruby#1778]
I'm unable to load the pg gem with master (19.3.0-dev) (rev. 9309ef1fce70b7c91cf913eee2741f43dc42a003), whereas I could load the gem with TruffleRuby 19.2.0.1:

```
❯ jt ruby --cexts-log-load -r pg -e 'puts $"'
Using Interpreted TruffleRuby: mxbuild/truffleruby-jvm
$ /home/nirvdrum/dev/workspaces/truffleruby-ws/truffleruby/mxbuild/truffleruby-jvm/jre/languages/ruby/bin/ruby \
  --experimental-options \
  --core-load-path=/home/nirvdrum/dev/workspaces/truffleruby-ws/truffleruby/src/main/ruby/truffleruby \
  --cexts-log-load \
  -r \
  pg \
  -e \
  'puts $"'

[ruby] INFO: loading cext implementation /home/nirvdrum/dev/workspaces/truffleruby-ws/truffleruby/mxbuild/truffleruby-jvm/jre/languages/ruby/lib/cext/libtruffleruby.so
[ruby] INFO: loading cext module /home/nirvdrum/dev/workspaces/truffleruby-ws/truffleruby/mxbuild/truffleruby-jvm/jre/languages/ruby/lib/gems/gems/pg-1.1.4/lib/pg_ext.so (requested as pg_ext)
[ruby] INFO: unsatisfied link error libtruffleruby.so: cannot open shared object file: No such file or directory
/home/nirvdrum/dev/workspaces/truffleruby-ws/truffleruby/mxbuild/truffleruby-jvm/jre/languages/ruby/lib/mri/rubygems/core_ext/kernel_require.rb:54:in `require': libtruffleruby.so: cannot open shared object file: No such file or directory (RuntimeError)
	from /home/nirvdrum/dev/workspaces/truffleruby-ws/truffleruby/mxbuild/truffleruby-jvm/jre/languages/ruby/lib/mri/rubygems/core_ext/kernel_require.rb:54:in `require'
	from /home/nirvdrum/dev/workspaces/truffleruby-ws/truffleruby/mxbuild/truffleruby-jvm/jre/languages/ruby/lib/gems/gems/pg-1.1.4/lib/pg.rb:4:in `<top (required)>'
	from /home/nirvdrum/dev/workspaces/truffleruby-ws/truffleruby/mxbuild/truffleruby-jvm/jre/languages/ruby/lib/mri/rubygems/core_ext/kernel_require.rb:131:in `require'
	from /home/nirvdrum/dev/workspaces/truffleruby-ws/truffleruby/mxbuild/truffleruby-jvm/jre/languages/ruby/lib/mri/rubygems/core_ext/kernel_require.rb:131:in `require'
	from -e:1:in `require'
	from -e:1:in `<main>'
```

**Comment by eregon** [comment]
Could you show:
```
ldd /home/nirvdrum/dev/workspaces/truffleruby-ws/truffleruby/mxbuild/truffleruby-jvm/jre/languages/ruby/lib/gems/gems/pg-1.1.4/lib/pg_ext.so

readelf -d /home/nirvdrum/dev/workspaces/truffleruby-ws/truffleruby/mxbuild/truffleruby-jvm/jre/languages/ruby/lib/gems/gems/pg-1.1.4/lib/pg_ext.so
```

For me that gives:
```
$ ldd /home/eregon/code/truffleruby-ws/truffleruby/mxbuild/truffleruby-jvm/jre/languages/ruby/lib/gems/gems/pg-1.1.4/lib/pg_ext.so
	linux-vdso.so.1 (0x00007fffa97c4000)
	libtruffleruby.so => /home/eregon/code/truffleruby-ws/truffleruby/mxbuild/truffleruby-jvm/jre/languages/ruby/lib/cext/libtruffleruby.so (0x00007f9d3ba58000)
...

readelf -d /home/eregon/code/truffleruby-ws/truffleruby/mxbuild/truffleruby-jvm/jre/languages/ruby/lib/gems/gems/pg-1.1.4/lib/pg_ext.so

Dynamic section at offset 0xee3b8 contains 28 entries:
  Tag        Type                         Name/Value
 0x000000000000001d (RUNPATH)            Library runpath: [/usr/lib64:/home/eregon/code/truffleruby-ws/truffleruby/mxbuild/truffleruby-jvm/jre/languages/ruby/lib/cext]
 0x0000000000000001 (NEEDED)             Shared library: [libtruffleruby.so]
 0x0000000000000001 (NEEDED)             Shared library: [libpolyglot-mock.so]
 0x0000000000000001 (NEEDED)             Shared library: [libpq.so.5]
 0x0000000000000001 (NEEDED)             Shared library: [libc.so.6]
...
```

This extra `/usr/lib64:` at the start of `Library runpath` sounds like a possible cause for this issue.

**Comment by nirvdrum** [comment]
Here you go:

```
❯ ldd /home/nirvdrum/dev/workspaces/truffleruby-ws/truffleruby/mxbuild/truffleruby-jvm/jre/languages/ruby/lib/gems/gems/pg-1.1.4/lib/pg_ext.so
	linux-vdso.so.1 (0x00007ffca28b1000)
	libtruffleruby.so => /home/nirvdrum/dev/workspaces/truffleruby-ws/truffleruby/mxbuild/truffleruby-jvm/jre/languages/ruby/lib/cext/libtruffleruby.so (0x00007f9865c7a000)
	libpolyglot-mock.so => not found
	libpq.so.5 => /usr/lib/x86_64-linux-gnu/libpq.so.5 (0x00007f9865c0a000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f9865a19000)
	libpolyglot-mock.so => not found
	libssl.so.1.1 => /usr/lib/x86_64-linux-gnu/libssl.so.1.1 (0x00007f9865985000)
	libcrypto.so.1.1 => /usr/lib/x86_64-linux-gnu/libcrypto.so.1.1 (0x00007f98656b0000)
	libgssapi_krb5.so.2 => /usr/lib/x86_64-linux-gnu/libgssapi_krb5.so.2 (0x00007f9865663000)
	libldap_r-2.4.so.2 => /usr/lib/x86_64-linux-gnu/libldap_r-2.4.so.2 (0x00007f986560d000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f98655ea000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f9865e29000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f98655e4000)
	libkrb5.so.3 => /usr/lib/x86_64-linux-gnu/libkrb5.so.3 (0x00007f9865505000)
	libk5crypto.so.3 => /usr/lib/x86_64-linux-gnu/libk5crypto.so.3 (0x00007f98654d4000)
	libcom_err.so.2 => /lib/x86_64-linux-gnu/libcom_err.so.2 (0x00007f98654cd000)
	libkrb5support.so.0 => /usr/lib/x86_64-linux-gnu/libkrb5support.so.0 (0x00007f98654be000)
	liblber-2.4.so.2 => /usr/lib/x86_64-linux-gnu/liblber-2.4.so.2 (0x00007f98654ad000)
	libresolv.so.2 => /lib/x86_64-linux-gnu/libresolv.so.2 (0x00007f9865492000)
	libsasl2.so.2 => /usr/lib/x86_64-linux-gnu/libsasl2.so.2 (0x00007f9865473000)
	libgssapi.so.3 => /usr/lib/x86_64-linux-gnu/libgssapi.so.3 (0x00007f986542f000)
	libgnutls.so.30 => /usr/lib/x86_64-linux-gnu/libgnutls.so.30 (0x00007f9865270000)
	libkeyutils.so.1 => /lib/x86_64-linux-gnu/libkeyutils.so.1 (0x00007f9865269000)
	libheimntlm.so.0 => /usr/lib/x86_64-linux-gnu/libheimntlm.so.0 (0x00007f986525d000)
	libkrb5.so.26 => /usr/lib/x86_64-linux-gnu/libkrb5.so.26 (0x00007f98651c9000)
	libasn1.so.8 => /usr/lib/x86_64-linux-gnu/libasn1.so.8 (0x00007f9865122000)
	libhcrypto.so.4 => /usr/lib/x86_64-linux-gnu/libhcrypto.so.4 (0x00007f98650ea000)
	libroken.so.18 => /usr/lib/x86_64-linux-gnu/libroken.so.18 (0x00007f98650d1000)
	libp11-kit.so.0 => /usr/lib/x86_64-linux-gnu/libp11-kit.so.0 (0x00007f9864f9c000)
	libidn2.so.0 => /usr/lib/x86_64-linux-gnu/libidn2.so.0 (0x00007f9864f7b000)
	libunistring.so.2 => /usr/lib/x86_64-linux-gnu/libunistring.so.2 (0x00007f9864df7000)
	libtasn1.so.6 => /usr/lib/x86_64-linux-gnu/libtasn1.so.6 (0x00007f9864de1000)
	libnettle.so.6 => /usr/lib/x86_64-linux-gnu/libnettle.so.6 (0x00007f9864da9000)
	libhogweed.so.4 => /usr/lib/x86_64-linux-gnu/libhogweed.so.4 (0x00007f9864d71000)
	libgmp.so.10 => /usr/lib/x86_64-linux-gnu/libgmp.so.10 (0x00007f9864cf0000)
	libwind.so.0 => /usr/lib/x86_64-linux-gnu/libwind.so.0 (0x00007f9864cc6000)
	libheimbase.so.1 => /usr/lib/x86_64-linux-gnu/libheimbase.so.1 (0x00007f9864cb2000)
	libhx509.so.5 => /usr/lib/x86_64-linux-gnu/libhx509.so.5 (0x00007f9864c64000)
	libsqlite3.so.0 => /usr/lib/x86_64-linux-gnu/libsqlite3.so.0 (0x00007f9864b41000)
	libcrypt.so.1 => /lib/x86_64-linux-gnu/libcrypt.so.1 (0x00007f9864b07000)
	libffi.so.6 => /usr/lib/x86_64-linux-gnu/libffi.so.6 (0x00007f9864afd000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f98649ac000)

❯ readelf -d /home/nirvdrum/dev/workspaces/truffleruby-ws/truffleruby/mxbuild/truffleruby-jvm/jre/languages/ruby/lib/gems/gems/pg-1.1.4/lib/pg_ext.so

Dynamic section at offset 0xef3a8 contains 29 entries:
  Tag        Type                         Name/Value
 0x000000000000001d (RUNPATH)            Library runpath: [/usr/lib/x86_64-linux-gnu:/home/nirvdrum/dev/workspaces/truffleruby-ws/truffleruby/mxbuild/truffleruby-jvm/jre/languages/ruby/lib/cext]
 0x0000000000000001 (NEEDED)             Shared library: [libtruffleruby.so]
 0x0000000000000001 (NEEDED)             Shared library: [libpolyglot-mock.so]
 0x0000000000000001 (NEEDED)             Shared library: [libpq.so.5]
 0x0000000000000001 (NEEDED)             Shared library: [libc.so.6]
 0x0000000000000007 (RELA)               0x59e0
 0x0000000000000008 (RELASZ)             4056 (bytes)
 0x0000000000000009 (RELAENT)            24 (bytes)
 0x000000006ffffff9 (RELACOUNT)          101
 0x0000000000000017 (JMPREL)             0x69b8
 0x0000000000000002 (PLTRELSZ)           8784 (bytes)
 0x0000000000000003 (PLTGOT)             0xee008
 0x0000000000000014 (PLTREL)             RELA
 0x0000000000000006 (SYMTAB)             0x2b8
 0x000000000000000b (SYMENT)             24 (bytes)
 0x0000000000000005 (STRTAB)             0x3e8c
 0x000000000000000a (STRSZ)              6989 (bytes)
 0x000000006ffffef5 (GNU_HASH)           0x2dd8
 0x0000000000000004 (HASH)               0x3154
 0x0000000000000019 (INIT_ARRAY)         0xef008
 0x000000000000001b (INIT_ARRAYSZ)       8 (bytes)
 0x000000000000001a (FINI_ARRAY)         0xef000
 0x000000000000001c (FINI_ARRAYSZ)       8 (bytes)
 0x000000000000000c (INIT)               0xebaf8
 0x000000000000000d (FINI)               0xebb14
 0x000000006ffffff0 (VERSYM)             0x2a48
 0x000000006ffffffe (VERNEED)            0x2d94
 0x000000006fffffff (VERNEEDNUM)         1
 0x0000000000000000 (NULL)               0x0
```

I have a slightly different path than you because I'm on Ubuntu. But it looks like the same case as you're seeing.

**Comment by eregon** [comment]
@rschatz fixed this in https://github.com/oracle/graal/commit/adc94cd5c47846bc9b4b9c242b61597b5f4ab4cf.
That fix will also be in 19.3.
